jdaviz/app.py

```diff
@@ -68,18 +68,30 @@
 class UnitConverterWithSpectral:
 
     def equivalent_units(self, data, cid, units):
+        # to debug
+        # print("eqv unit cid : ", cid, ", units : ", units)
+
         if cid.label == "flux":
             eqv = u.spectral_density(1 * u.m)  # Value does not matter here.
             list_of_units = set(list(map(str, u.Unit(units).find_equivalent_units(
-                include_prefix_units=True, equivalencies=eqv))) + [
-                    'Jy', 'mJy', 'uJy',
+                include_prefix_units=True, equivalencies=eqv)))
+                + [
+                    'Jy', 'mJy', 'uJy', 'MJy',
                     'W / (m2 Hz)', 'W / (Hz m2)',  # Order is different in astropy v5.3
```

Comment:
do we know which version is for 5.3+?  #2827 will bump astropy requirement beyond 5.3 so maybe at that point we can clean this up?

Reply:
I've found the order with which units are displayed/stored has significancy in the conversion and in causing tracebacks, I would like this to also to be cleaner. The "W / (Hz m^2)" is the unit 5.3+.

Reply:
nothing to do here for now.  Once rebased on top of this, #2827 can always remove the unneeded line.